File: 0683/Sesión 05 - lib C#/Publicaciones_v00/Publicaciones_v00/Autor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Publicaciones_v00
{ 

    internal class Autor
    {
        private string apellidos;
        private string nombres;

        public Autor(string nombre_completo_autor)
        {
            int posicion_coma = nombre_completo_autor.IndexOf(",");
            this.apellidos = nombre_completo_autor.Substring(0, posicion_coma).Trim();
            this.nombres = nombre_completo_autor.Substring(++posicion_coma).Trim();
        }
        public override string ToString()
        {
            string copia_nombre = this.nombres;
            string abreviatura = "";
            int posicion_blanco;
            do
            {
                posicion_blanco = copia_nombre.IndexOf(" ");
                if (!abreviatura.Equals(""))
                    abreviatura += " ";
                abreviatura += copia_nombre.Substring(0, 1);
                abreviatura += ".";
                if (posicion_blanco != -1)
                    copia_nombre = copia_nombre.Substring(posicion_blanco + 1).Trim();
            }
            while (posicion_blanco != -1);
            return abreviatura + " " + this.apellidos;
        }
    }
}

File: 0683/Sesión 04 - lib java/Publicaciones_v00/src/main/java/pe/edu/pucp/inf30/sesiones/publicaciones00/publicaciones/Autor.java
package pe.edu.pucp.inf30.sesiones.publicaciones00.publicaciones;

public class Autor {

    private String apellidos;
    private String nombres;

    public Autor(String nombre_completo_autor) {
        Integer posicion_coma = nombre_completo_autor.indexOf(",");
        this.apellidos = nombre_completo_autor.substring(0, posicion_coma).trim();
        this.nombres = nombre_completo_autor.substring(posicion_coma + 1).trim();
    }

    @Override
    public String toString() {
        String copia_nombre = this.nombres;
        String abreviatura = "";
        Integer posicion_blanco;
        do {
            posicion_blanco = copia_nombre.indexOf(" ");
            if (!abreviatura.equals("")) {
                abreviatura = abreviatura.concat(" ");
            }
            abreviatura = abreviatura.concat(copia_nombre.substring(0, 1));
            abreviatura = abreviatura.concat(".");
            if (posicion_blanco != -1) {
                copia_nombre = copia_nombre.substring(posicion_blanco + 1).trim();
            }
        } while (posicion_blanco != -1);
        return abreviatura.concat(" ").concat(this.apellidos);
    }
}
